src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -512,6 +512,50 @@ LoadShardIntervalList(Oid relationId)
 }
 
 
+/*
+ * LoadShardIntervalWithLongestShardName is a utility function that returns
+ * the shard interaval with the largest shardId for the given relationId. Note
+ * that largest shardId implies longest shard name.
+ */
+ShardInterval *
+LoadShardIntervalWithLongestShardName(Oid relationId)
+{
+	CitusTableCacheEntry *cacheEntry = GetCitusTableCacheEntry(relationId);
+	int maxShardIndex = cacheEntry->shardIntervalArrayLength - 1;
+	uint64 largestShardId = INVALID_SHARD_ID;
+
+	if (cacheEntry->hasUniformHashDistribution)
+	{
+		ShardInterval *shardInterval =
+			(cacheEntry->sortedShardIntervalArray[maxShardIndex]);
+
+		largestShardId = shardInterval->shardId;
+	}
+	else
+	{
+		int shardIndex = 0;
+		for (; shardIndex <= maxShardIndex; ++shardIndex)
+		{
+			ShardInterval *currentShardInterval =
+				cacheEntry->sortedShardIntervalArray[shardIndex];
+
+			if (largestShardId < currentShardInterval->shardId)
+			{
+				largestShardId = currentShardInterval->shardId;
+			}
+		}
+	}
+
+	/* this cannot happen, still be defensive */
+	if (largestShardId == 0)
+	{
+		ereport(ERROR, (errmsg("unexpected shardId: %lu", largestShardId)));
```

Comment:
append-distributed tables can have 0 shards

Reply:
as far as I see, we already don't allow partitioning append distributed tables, so do you think that error message is sufficient @marcocitus  ?

>SELECT create_distributed_table('measurement','city_id','append');
ERROR:  distributing partitioned tables in only supported for hash-distributed tables